In [1]:
# The code was removed by Watson Studio for sharing.

# Merging with Geospatial Data 

## In this notebook, I will merge the data that we scraped from the Toronto Postal Code Wikipedia page with the geospatial data in the CSV file provided as it is more reliable that the geocoder package.

##### Let's start by reading the csv files we created and convert them into Pandas DataFrames

In [3]:
#Getting csv file and loading as pandas datafarme
df1= pd.read_csv(project.get_file('Geospatial_Coordinates.csv'))
df2= pd.read_csv(project.get_file('toronto.csv'))

##### Now let's take a look at our new dataframes

In [4]:
df1.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [5]:
df2.head()

Postcode      Borough                          Neighborhood
0      M1B  Scarborough                         Rouge,Malvern
1      M1C  Scarborough  Highland Creek,Rouge Hill,Port Union
2      M1E  Scarborough       Guildwood,Morningside,West Hill
3      M1G  Scarborough                                Woburn
4      M1H  Scarborough                             Cedarbrae

#### Now let's take confirm the shapes of our dataframes

In [7]:
print('df1: ',df1.shape)
print('df2: ', df2.shape)

df1:  (103, 3)
df2:  (103, 3)


### We can add the columns from our first dataframe with the Toronto Postcode coordinates with our second dataframe that contains the neighborhoods of Toronto.

In [13]:
df2['Latitude'] = df1['Latitude']
df2['Longitude'] = df1['Longitude']

df2.head()

Postcode      Borough                          Neighborhood   Latitude  \
0      M1B  Scarborough                         Rouge,Malvern  43.806686   
1      M1C  Scarborough  Highland Creek,Rouge Hill,Port Union  43.784535   
2      M1E  Scarborough       Guildwood,Morningside,West Hill  43.763573   
3      M1G  Scarborough                                Woburn  43.770992   
4      M1H  Scarborough                             Cedarbrae  43.773136   

   Longitude  
0 -79.194353  
1 -79.160497  
2 -79.188711  
3 -79.216917  
4 -79.239476

### Now let's confirm the shape on our new dataframe which should be 103 rows with 5 columns.

In [14]:
df2.shape

(103, 5)

## Now that we have successfully merged our data, let's save the new dataframe to our storage.

In [ ]:
# Save dataframe as csv file to storage
project.save_data(data=df2.to_csv(index=False),file_name='toronto_geo.csv',overwrite=True)